java/ql/lib/semmle/code/java/frameworks/spring/SpringWebMethod.qll

```diff
@@ -0,0 +1,30 @@
+/** Provides models of taint flow in `org.springframework.web.method` */
+
+import java
+private import semmle.code.java.dataflow.ExternalFlow
+
+// currently only models classes in the `support` subpackage
+private class FlowSummaries extends SummaryModelCsv {
+  override predicate row(string row) {
+    row =
+      [
+        // for review: arguably this shouldn't be modeled as the implementations of resolveArgument that I've seen are effectively sanitized
+        "org.springframework.web.method.support;HandlerMethodArgumentResolver;true;resolveArgument;;;Argument[2];ReturnValue;taint",
```

Comment:
Can you give an example what sanitisation you've seen? If the user provides a form value or whatever is going to be associated with the given method parameter, will it be rendered safe for pasting into HTML output? Would it still work as SQL injection for example?

Reply:
In the two cases I saw in Spring data, the user-input strings were parsed into a `Sort` object, and page size and page number, neither of which I think can carry meaningful taint.

Reply:
Ah sure, but presumably this would propagate taint when the argument type isn't restrictive like that? If so I'd say we should regard it as a taint propagator in general and sanitize depending on the particular usage (e.g. integers are already blanket sanitized by the XSS query)

Reply:
Yeah, that was my thinking. I'll leave this in then.